src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.spring.controller;
+
+import java.util.NoSuchElementException;
+import mate.academy.spring.mapper.impl.response.UserResponseMapper;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    private final UserService userService;
+    private final UserResponseMapper userResponseMapper;
+
+    public UserController(UserService userService, UserResponseMapper userResponseMapper) {
+        this.userService = userService;
+        this.userResponseMapper = userResponseMapper;
+    }
+
+    @GetMapping("/by-email")
```

Comment:
It is better to use @RequestParam instead of @PathVariable for getting email parameter from the request.